graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -307,24 +334,34 @@ def _process_directive_definition(directive, existing_directives, merged_schema_
                 )
             )
     # new directive
-    merged_schema_ast.definitions.append(directive)
-    existing_directives[directive_name] = directive
+    new_definitions = list(merged_schema_ast.definitions)
+    new_definitions.append(directive)
+    new_merged_schema_ast = ast_types.DocumentNode(definitions=new_definitions)
+    new_existing_directives = deepcopy(existing_directives)
```

Comment:
This should be able to be safely replaced with just `copy()` (which in this case does the same thing as `dict(existing_directives)` so feel free to use that if that's easier) since we don't make changes within keys or values of the dict — we only add a new key.